### ISSUE TITLE
FAB - Missing conversations in recent list

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#21283]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Preconditions: user has already navigated to NewDot and logged in with an account that has a lot of conversations.

1. Open http://staging.new.expensify.com/
2. Log in [angela.puella@mail.com](mailto:angela.puella@mail.com)
1. Click on any chat on the LHN that is not the most recent one
2. Click on the "Global action" button (Big green plus button in LHN)
3. Click on "New chat" / "New group"

## Expected Result:
The search list should be divided by "Recents" and "Contacts".
There should be 5 conversations displayed in "Recents".

## Actual Result:
There's only a conversation displayed in "Recents"

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.30.5

**Reproducible in staging?:** Yes

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/9710ed46-def9-4696-ad4e-53f2b081cb2a

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by situchan** [comment]
I think this is backend issue. Happening on production as well.
To reproduce, logout (to clear all stale contacts data) and re-login

**Comment by WoLewicki** [comment]
For me, those recent contacts don't change at all when navigating, and they should show the top 5 most recent opened reports?

**Comment by situchan** [comment]
Here's the root cause
Backend doesn't return `login` in `personalDetails` so it fails this condition:
https://github.com/Expensify/App/blob/2b2f4979eb61abfbdffb93c61d0e5814421d826c/src/libs/OptionsListUtils.js#L608C16-L612

We can fix this by replacing `detail.login` by `detail.accountID` but I think this isn't enough fix.

**Comment by mountiny** [comment]
Thanks for linking, I also dont think this is a deploy blocker, discussing in the [slack thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1687390694872859)

**Comment by Skalakid** [comment]
I'm Michał from Software Mansion and I would like to work on this issue.

**Comment by luacmartins** [comment]
Removing the blocker label - see https://expensify.slack.com/archives/C01GTK53T8Q/p1687390694872859

**Comment by Skalakid** [comment]
After discussing the solution in the Slack thread we decided to wait for [this plan](https://expensify.slack.com/archives/C01GTK53T8Q/p1687527704008849?thread_ts=1687390694.872859&cid=C01GTK53T8Q). Currently it is hard to fix this bug to work online and offline because we don't have needed user logins data from the server.

**Comment by Skalakid** [comment]
Not overdue, still discussing it on slack

**Comment by Skalakid** [comment]
It seems it is a backend issue since we don't receive the necessary data about all users. The logins/emails that we need to represent user items in the `Recent` list are unavailable. It is due to the fact that user data is limited until we enter a conversation with somebody. After this, the user will be "known" by us, and we will have more detailed user data.

The same problem appears in the `Contacts` list. To solve this issue we have to find a way how to get detailed data about all our contacts. With it we can easily create `Recent` list with the current behaviour

**Comment by Skalakid** [comment]
@mountiny I completely lost notification about this issue. Like I wrote above I think the problem was on the backend side. Do you know if it was solved?

**Comment by mountiny** [comment]
Yeah I think this one has not been reproduced for a while, if it would be repro again, we can reopen

**Comment by lanitochka17** [comment]
Issue is still reproducible on the latest build 9.0.85-3

https://github.com/user-attachments/assets/7c8a6a01-0183-491b-a6fd-f323a369facc

![Image](https://github.com/user-attachments/assets/0fd0febd-7859-417b-9c58-83d859721c87)

**Comment by izarutskaya** [comment]
Created new issue for this [comment](https://github.com/Expensify/App/issues/21283#issuecomment-2590987506) - new GH https://github.com/Expensify/App/issues/55266
